test/cli/uninitialized-explanation/uninitialized-explanation.out

```diff
@@ -0,0 +1,18 @@
+test/cli/uninitialized-explanation/uninitialized.rb:15: Expected `Integer` but found `T.nilable(Integer)` for argument `x` https://srb.help/7002
+    15 |  takes_int(x)
+          ^^^^^^^^^^^^
+    test/cli/uninitialized-explanation/uninitialized.rb:5: Method `Object#takes_int` has specified `x` as `Integer`
+     5 |sig {params(x: Integer).void}
+                    ^
+  Got T.nilable(Integer) originating from:
+    test/cli/uninitialized-explanation/uninitialized.rb:10:
+    10 |    x = 1
+                ^
+    test/cli/uninitialized-explanation/uninitialized.rb:8: Type may be `NilClass` due to uninitialized variables in:
+     8 |def uninitialized_loc_is_decl_loc
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Comment:
> I'm not sure I'm happy with this phrasing, FWIW.

What about just

`    foo.rb:8: Potentially uninitialized (`NilClass`)`

?

> It would be nice to name the specific variable that's uninitialized, but I don't think we have that information at this point.

I think it's already mentioned in the error above (``for argument `x` ``)


> Also, I chose the plural `variables` here because I _think_ it's possible for more than one uninitialized variable to be part of the origins, but not sure if I like it.

If that's the case, could you add a test? My gut tells me that a `TypeAndOrigins` tracks the type of a single variable in the environment, along with all the bindings it participated in that affect its type (e.g., a `TypeAndOrigins` structure has one type (because it's one variable) but many origins (because that one variable can be read/written in many places).

Reply:
> `foo.rb:8: Potentially uninitialized (`NilClass`)`

Yeah, that sounds better. I think I will stick an `in:` on the end since we are still displaying carets for the loc.

> I think it's already mentioned in the error above (for argument `x` )

That's true! I guess I wasn't sure under what circumstances the "uninitialized" message might leak into more complicated expressions, though. I was thinking something like `x+y` where `x` and `y` are both potentially uninitialized... but I guess actually in that case there will be two distinct errors for the arguments `x` and `y`, not one error for the expression `x+y`. (So, in other words, I don't actually have a concrete example!)

> If that's the case, could you add a test?

Yup, will play with it and see if I can find an example that is confusing in the way I was imagining. (If not, there's probably no issue here.) This should address the FUD I was feeling on the previous point as well.

Reply:
Just to close this out:

> My gut tells me that a `TypeAndOrigins` tracks the type of a single variable in the environment...

Yup, I had it in my head that `getTypeAndOrigin` was coming from an "`Expr`", but we're in CFG-land at this point so that isn't even a thing, and in reality it comes from a `LocalRef`. So cases like I was imagining don't exist.